### ISSUE TITLE
Error message the first time I am entering "update" in A1

### ISSUE THREAD

**Issue by marinaluchner** [botcs/wolfson-scheduler#2]
Hi, 

I am happily using this scheduler now and am very grateful for its existence. I am getting the following error message every time I am typing "update" into A1 in the spreadsheet. The second time around it suddenly works.

Here is the error message:
[2024-01-27 13:12:02,695][ERROR][server.py:503 - <module>()] Traceback (most recent call last):
  File "/mnt/c/Users/magd5549/OneDrive - Nexus365/Rowing/wolfson-scheduler-main/server.py", line 500, in <module>
    main(args)
  File "/mnt/c/Users/magd5549/OneDrive - Nexus365/Rowing/wolfson-scheduler-main/server.py", line 482, in main
    launch_periodic_trigger(spreadsheet=spreadsheet, time_interval=3)
  File "/mnt/c/Users/magd5549/OneDrive - Nexus365/Rowing/wolfson-scheduler-main/server.py", line 449, in launch_periodic_trigger
    if trigger_value.lower() == "update":
       ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'lower'

Very best, LBC captain

**Comment by botcs** [comment]
Hi,

I'm glad to hear the tool has been useful :) thanks for the feedback.

Regarding the error message, the following could be the case: when gspread is trying to read from a field that is empty, by default it returns None.

Try to see if the update message is actually written into the default A1 field.
Maybe there's some weird behavior coming from merging the cells.

If you could share a copy of the sheet with me (with names potentially obfuscated) I could take a look into it.

(Reached out in email)

Thanks,
Csabi

**Comment by botcs** [comment]
[UPDATE]
Hi @marinaluchner,

tl;dr: found the bug, it was a wrong assumption about UI, fixed it in 8f8a5e0.

here's the detailed report:
I've gone through the entire installation process with a clone of the provided sheet.
For installing the dependencies, I used the provided modified environment file and the original one as a control.
Both the original and the modified `environment.yml` installs without errors from a clean installation of `miniconda -> Latest - Conda 23.11.0 Python 3.11.5 released December 20, 2023` at the time of writing.

Both on Linux (Ubuntu 20.04) and MacOS (Ventura 13.0.1) I was able to reproduce the error by following the demonstrated usage. 

The detailed reason:
When you modify the cell content, the browser client only sends the data from your computer to the google servers when the editing is completed.
If you press delete / backspace the content, then the new updated value is empty (which was a case that was not handled in the codebase, now that bit is fixed).
Once you type "update" it only gets registered in the cloud if you press enter / switch focus away from the window.
Now in our demo, when you typed update, the focus was still on on the cell, and only when you switched to the terminal did we see the miraculously disappearing bug.
I managed to reproduce this pattern on both the Linux and the MacOS versions, and now it's fixed on both:
it will only start updating once you actually finalise the cell editing - by hitting enter, clicking away, swapping windows etc.

Thanks for helping hunting down this corner case!
Csabi
